fe/fe-core/src/main/java/com/starrocks/planner/JDBCScanNode.java

```diff
@@ -144,6 +159,93 @@ private void createJDBCTableFilters() {
         }
     }
 
+    private static final Pattern SESSION_VARIABLE_PATTERN = Pattern.compile(
+            "(@'[^']+'|@\"[^\"]+\"|@[A-Za-z0-9_]+|[A-Za-z0-9_]+)\\s*=\\s*('[^']*'|\"[^\"]*\"|\\d+)(?:,|$)"
+    );
+
+    private void createJDBCTableSessionVariableHints() {
+        String jdbcExternalTableSessionVariables =
+                ConnectContext.get().getSessionVariable().getJdbcExternalTableSessionVariables();
+        if (StringUtils.isEmpty(jdbcExternalTableSessionVariables)) {
+            return;
+        }
+        if (!isMysql()) {
+            throw new UnsupportedOperationException(
+                    String.format("Sending session variable to JDBC external table is only supported for MYSQL protocol")
+            );
+        }
+        boolean isStarRocksTable = isStarRocksTable();
+
+        // Prepare clauses for hints
+        StringBuilder setVarHintClause = new StringBuilder();
+        StringBuilder setUserVariableHintClause = new StringBuilder();
+
+        // Split the input into 'variableName=variableValue' string assignments and iterate over each assignment
+        String[] sessionVariableAssignments =
+                jdbcExternalTableSessionVariables.split(",(?=(?:[^']*'[^']*')*[^']*$)");
+        for (String assignment : sessionVariableAssignments) {
```

Comment:
COuld you refactor this code a little bit in following way? 
- first collect user variables and hint clauses in the for loop.
- then assembe them into a string later.